### PR TITLE
App specific notif and invit

### PR DESCRIPTION
**Pull request by Poppe-A**
#6989

### CODE REVIEW

File: libs/model/src/lib/notification.ts [hunk 1]
```diff
@@ -31,43 +44,49 @@ export const notificationTypesBase = [
...
 
 export type NotificationTypesBase = typeof notificationTypesBase[number];
+export type NotificationTypesFestival = typeof festivalSpecificNotifications[number];
+export type NotificationTypesCatalog = typeof catalogSpecificNotifications[number];
 type NotificationTypesPlus = typeof notificationTypesPlus[number];
-export type NotificationTypes = NotificationTypesBase | NotificationTypesPlus;
+export type NotificationTypes = NotificationTypesBase | NotificationTypesPlus | NotificationTypesFestival | NotificationTypesCatalog;
+
+export function isCatalogNotification(notification: Notification, app: App) {
+  if (app === 'catalog') {
```
Comment (RemcoSimonides) [inline libs/model/src/lib/notification.ts:72]: Why do you check for app here? It depends on the notification.type, no?

File: libs/model/src/lib/notification.ts [hunk 2]
```diff
@@ -31,43 +44,49 @@ export const notificationTypesBase = [
...
+    return catalogSpecificNotifications.includes(notification.type as NotificationTypesCatalog) 
+      || notification.type === 'movieAccepted' && notification._meta.createdFrom === 'catalog'
+      || notification.type === 'movieSubmitted';
```
Comment (RemcoSimonides) [inline libs/model/src/lib/notification.ts:75]: Something is going wrong here and also it doesn't look like a scalable solution

File: libs/model/src/lib/notification.ts (hunk 1 above)
Comment (Poppe-A) [reply libs/model/src/lib/notification.ts:72]: Yup but I need to know the current app to display relative notif only

File: libs/model/src/lib/notification.ts [hunk 3]
```diff
@@ -8,66 +8,66 @@ import { StorageFile } from './media';
...
-  // Notifications relative to invitations
-  'requestFromUserToJoinOrgPending', // Notification sent to the user that made the request
-  'invitationToJoinOrgDeclined',
-
-  // Other notifications
-  'movieSubmitted', // (catalog only)
-  'organizationAcceptedByArchipelContent',
-  'orgAppAccessChanged',
-  'userRequestAppAccess',
+export const notifications = {
```
Comment (RemcoSimonides) [inline libs/model/src/lib/notification.ts:11]: Add
```
financiers: [] as const,
crm: [] as const,
```
So it's equal to App type and you don't need to do `as 'catalog' | 'festival'` later

File: libs/user/src/lib/auth/forms/notifications-form/notifications-form.component.ts [hunk 4]
```diff
@@ -105,7 +106,7 @@ const tables: { title: string, types: string[], appAuthorized: App[] }[] = [
 })
 export class NotificationsFormComponent {
 
-  public types = [...notificationTypesBase];
+  public types = [...notificationTypes];
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/notifications-form/notifications-form.component.ts:109]: Why not just `public types = notificationTypes;`?

File: libs/model/src/lib/notification.ts [hunk 5]
```diff
@@ -8,66 +8,66 @@ import { StorageFile } from './media';
...
+export type NotificationTypes = typeof notifications.catalog[number] | typeof notifications.festival[number] | typeof notifications.shared[number];
+export const notificationTypes = [...notifications.festival, ...notifications.catalog, ...notifications.shared];
+type AppNotificationType = typeof notifications;
+function isNotificationType<K extends keyof AppNotificationType>(
+  type: NotificationTypes,
+  kind: K
+): type is AppNotificationType[K][number] {
+  const notif: string[] = notifications[kind] as any;
+  return notif.includes(type);
+}
```
Comment (RemcoSimonides) [inline libs/model/src/lib/notification.ts:66]:
```suggestion
type AppNotificationType = typeof notifications;
type AppNotificationKey = keyof AppNotificationType;
export type NotificationTypes = typeof notifications[AppNotificationKey][number];
export const notificationTypes = [...notifications.festival, ...notifications.catalog, ...notifications.shared];
function isNotificationType<K extends keyof AppNotificationType>(
  type: NotificationTypes,
  kind: K
): type is AppNotificationType[K][number] {
  const notif: string[] = notifications[kind] as any;
  return notif.includes(type);
}
```

File: libs/notification/src/lib/+state/notification.service.ts [hunk 6]
```diff
@@ -46,6 +47,7 @@ export class NotificationService extends CollectionService<NotificationState> {
     switchMap((notifications) => {
       const promises = notifications
         .filter(n => n.app?.isRead !== undefined)
+        .filter(n => isAppNotification(n.type, this.app as 'catalog' | 'festival'))
```
Comment (GrandSchtroumpf) [inline libs/notification/src/lib/+state/notification.service.ts:50]: How do you know it's not `financiers` ?

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**